java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java

```diff
@@ -351,11 +351,11 @@ private void buildRow() throws SQLException {
                 }
             }
             if (!ok) {
-                throw new SQLException("append failed");
+                throw new SQLException("append failed on currentDataIdx{curType}, current data"+data);
```

Comment:
plz print the real `currentDataIdx` and `curType`, the msg in issue is just pseudocode

Reply:
curType is defined in "else block",  It has no scope outside that.

Reply:
you can move `DataType curType` or just use `currentDatasType.get(currentDataIdx)`